internal/app/machined/pkg/controllers/k8s/internal/nodewatch/nodewatch.go

```diff
@@ -92,11 +92,25 @@ func (r *NodeWatcher) Watch(ctx context.Context, logger *zap.Logger) (<-chan str
 
 	informerFactory.Start(ctx.Done())
 
-	logger.Debug("waiting for node cache sync")
+	go func() {
+		logger.Debug("waiting for node cache sync")
 
-	informerFactory.WaitForCacheSync(ctx.Done())
+		result := informerFactory.WaitForCacheSync(ctx.Done())
 
-	logger.Debug("node cache sync done")
+		var synced bool
+
+		// result should contain a single entry
+		for _, v := range result {
+			synced = v
+		}
```

Comment:
This is probably

```suggestion
		for _, v := range result {
			synced |= v
		}
```
or

```suggestion
		for _, v := range result {
			synced &= v
		}
```

or even

```suggestion
		for _, v := range result {
			synced = v
			break
		}
```

But I'm not sure if there any reason to get random element of the map by iterating across the whole map.

Reply:
if you read the comment above, the map contains a single element